### DISCUSSION TITLE
Rendering MSDF.

### DISCUSSION THREAD

**Discussion by JulienGery** [Chlumsky/msdfgen#183] [Q&A] [answered]
Hello! I hope you're doing well. 
I'm resuming work on my project, which aims to render MSDF on the GPU using Vulkan. Currently, I'm facing challenges with the "last part". I can't render them properly. Here is an example:
YOUR OUTPUT:
![output](https://github.com/Chlumsky/msdfgen/assets/75223846/7d164fad-5612-4f75-a5ee-cd33d9de7dc5)
MY OUTPUT:
![output](https://github.com/Chlumsky/msdfgen/assets/75223846/6917095a-8fdf-40bd-ae94-5766b5fa6e9a)

While the colors are not identical, I would say they appear to be Somewhat similar.
The real issue occurs when i try to render them:
Yours renders perfectly (obviously), whereas mine (using your values) turns out all white, and I'm unsure why.
Trying another value for the pixel range (0.002) yield this result: 
![render](https://github.com/Chlumsky/msdfgen/assets/75223846/def41d6d-2e04-49c8-89ea-0b02fbd5647c)

Which is quite unappealing and has a strange fog. Additionally, cyan lacks anti-aliasing, while purple and yellow have it?

From what I understand, this issue seems to be related to my anti-aliasing.
I'm implementing it using a function in the form of ax + 0.5, where x represents a clamped signed distance, and 'a' is a coefficient determined by the resolution.

So, in essence, here are my questions:

What does "pixel range" refer to within the render-sdf function?
What does "mid value" mean in the context of the render-sdf function?
Can such an MSDF with certain modifications be rendered correctly, or have I been approaching this incorrectly from the beginning?

**Comment by Chlumsky** [comment]
I don't understand what you mean. When I apply the median on your image I get this.

![image](https://github.com/Chlumsky/msdfgen/assets/18639794/ad691e26-dc45-404f-84d8-883d604fe730)

**Reply by JulienGery** [reply]
Hi! It's quite relieving to witness it rendering correctly.
The issue appears to be linked to my conversion between my format (uint32_t) and yours (bitmap<float, 3>). Do you store the color values in an RGB format, with each value ranging from 0 to 255, or was I mistaken? Additionally, which values did you use for the rendering?

**Reply by JulienGery** [reply, marked as answer]
I found that the issue was with my mid value; it should have been .5 instead of -.5. Additionally, I realized I needed to divide my color value by 255.
